Gordon360/Services/LostAndFoundService.cs

```diff
@@ -265,52 +237,63 @@ public async Task UpdateMissingItemReportAsync(int missingItemID, MissingItemRep
 
             var original = await context.MissingReports.FindAsync(missingItemID);
 
             if (original == null)
             {
                 throw new ResourceNotFoundException() { ExceptionMessage = "The Missing Item Report was not found" };
             }
-
-            if (original.submitterID == idNum || hasFullPermissions(username))
-            {
-                original.status = status;
-
-                await context.SaveChangesAsync();
-            }
-            else
+            
+            // If a non-admin user attempts to update the status of a report
+            if (original.submitterID != idNum && !hasFullPermissions(username))
             {
                 throw new UnauthorizedAccessException("Cannot modify a report that doesn't belong to you!");
             }
+
+            original.status = status;
+
+            await context.SaveChangesAsync();
         }
 
         /// <summary>
         /// Get the list of missing item reports for given user.
         /// </summary>
-        /// <param name="username">The username of the person making the request</param>
+        /// <param name="requestedUsername">The username to get the data of, if allowed</param>
+        /// <param name="requestorUsername">The username of the person making the request</param>
         /// <returns>an Enumerable of Missing Item Reports containing all missing item reports</returns>
-        public IEnumerable<MissingItemReportViewModel> GetMissingItems(string username)
+        /// <exception cref="ResourceNotFoundException">If a user requests reports they are not permitted to access</exception>
+        public IEnumerable<MissingItemReportViewModel> GetMissingItems(string requestedUsername, string requestorUsername)
         {
-            return context.MissingItemData
-                .Where(x => x.submitterUsername == username && !x.forGuest)
-                .GroupJoin(context.ActionsTakenData
-                .Where(x => x.isPublic),
-                    missingItem => missingItem.ID,
-                    action => action.missingID,
-                    (missingItem, action) => MissingItemReportViewModel.From(missingItem, action));
+            if (hasFullPermissions(requestorUsername))
+            {
+                // If this is an admin user, get the reports for the requested username
+                return context.MissingItemData
+               .Where(x => x.submitterUsername == requestedUsername && !x.forGuest)
+               .GroupJoin(context.ActionsTakenData
+               .Where(x => x.isPublic),
+                   missingItem => missingItem.ID,
+                   action => action.missingID,
+                   (missingItem, action) => MissingItemReportViewModel.From(missingItem, action));
+            }
+            else
+            {
+                // If a non-admin user requests the reports of someone else
+                if (requestedUsername.ToLower() != requestedUsername.ToLower())
```

Comment:
I think you accidentally used the same username twice. Also, C# has a more efficient way to compare strings case-insensitive, using the `StringComparison` argument to `String.Equals`:
```suggestion
                if (requestorUsername.Equals(requestedUsername, StringComparison.OrdinalIgnoreCase))
```